DeferredTests/DeferredTests.swift

```diff
@@ -305,4 +320,63 @@ class DeferredTests: XCTestCase {
         waitForExpectationsWithTimeout(0.5, handler: nil)
     }
     
+    func testIsFilledCanBeCalledMultipleTimesNotFilled() {
+        let d = Deferred<Int>()
+        XCTAssertFalse(d.isFilled)
+        XCTAssertFalse(d.isFilled)
+        XCTAssertFalse(d.isFilled)
+    }
+
+    func testIsFilledCanBeCalledMultipleTimesWhenFilled() {
+        let d = Deferred<Int>(value: 42)
+        XCTAssertTrue(d.isFilled)
+        XCTAssertTrue(d.isFilled)
+        XCTAssertTrue(d.isFilled)
+    }
+
+    // The QoS APIs do not behave as expected on the iOS Simulator, so we only
+    // run these tests on real devices. This check isn't the most future-proof;
+    // if there's ever another archiecture that runs the simulator, this will
+    // need to be modified.
+    //
+    // TODO: Add a clause for a tvOS test target
+    #if os(OSX) || (os(iOS) && !(arch(i386) || arch(x86_64)))
```

Comment:
Simple `#define`s from C are only available as `let` symbols in Swift; they are not available for compile-time checking. We might be able to define a convoluted `#if` flag using build configurations, though @jgallagher and I determined when this test was written for internal code that this was the best solution.